app/assets/javascripts/discourse/tests/helpers/create-pretender.js

```diff
@@ -39,7 +40,15 @@ const helpers = { response, success, parsePostData };
 
 export let fixturesByUrl;
 
-export default new Pretender();
+const instance = new Pretender();
+
+const oldRegister = instance.register;
+instance.register = (...args) => {
```

Comment:
_(copied from https://github.com/discourse/discourse/pull/12837)_

Not a huge fan of this hack, but setting `baseUri` in `lib/get-url` (which is needed to get the right CDN URL) makes our `ajax` lib prepend the subfolder path to all the requests it makes which makes pretender unable to recognize our fixtures. One alternative fix is to add an `isTesting()` check in our `ajax` lib [here](https://github.com/discourse/discourse/blob/b32c7df0e9608863d2ff78b7823d636350a69428/app/assets/javascripts/discourse/app/lib/ajax.js#L169) and if we're in test mode then we skip the `getURL` call, but I'm not a fan of that approach either.

Reply:
I don't quite understand this change. Pretender assumes there is no subfolder because our tests do that. Are there some pretender tests that you'd like to work with subfolder?

Reply:
Before the original PR was reverted, I tried running a theme's tests on Meta but the tests kept freezing and never finished. After some debugging I realized that it was hanging because at some point during the tests Discourse tried to fetch `jquery.magnific-popup.min.js` (which is lazy-loaded) but it failed with 404 because it hit `meta.discourse.org` instead of the configured CDN.

Now that I've thought this a little bit more, I see we have 3 solutions:

1. Configure CDN when running theme tests so that lazy-loaded assets are loaded exactly like how we do for the rest of the app, but this means we have to also configure subfolder (because getting the right CDN URL requires the subfolder bit). This is what I've done here, but configuring subfolder will make our `ajax` prepend subfolder path to requests paths and that breaks pretender (because it would no longer recognize e.g. `/subfolder/latest.json`)

2. In test mode we ignore the CDN and load directly from the app. E.g., instead of loading from `https://d3bpeqsaub0i6y.cloudfront.net/javascripts/magnific-popup/1.1.0/jquery.magnific-popup.min.js` we load directly from meta `https://meta.discourse.org/javascripts/magnific-popup/1.1.0/jquery.magnific-popup.min.js`.

3. Don't lazy-load anything in tests and include everything tests could possibly need up front. This means we need to add an `isTesting` check in `loadScript` and if we're in test mode then we don't attempt to load anything.